### DISCUSSION TITLE
Manage logfile

### DISCUSSION THREAD

**Discussion by hotting** [ZuinigeRijder/SolisCloud2PVOutput#22] [Ideas]
The logging works perfectly.

Only one point of improvement is the growth of the log file. A log file is generally only used in special circumstances that do not occur very often. The log file can therefore grow unnoticed to an undesired size.

To prevent this, every daily start of the script can be checked whether the log file exceeds a certain size and then halve it, for example. This way you can always have a number of days of log available by choosing the right size.

Another possibility is to let the logfile rotate in the logrotate of the raspberry. This makes the script less easy to install as it requires specific knowledge of the logrotate system

A suggestion for the first option is:

```
import os

# Set the path to the file
file_path = 'solis.log'

# Check if the file exists and is larger than 1MB
if os.path.exists(file_path) and os.path.getsize(file_path) > 1024*1024:
    # Open the file in read mode and get its size
    with open(file_path, 'r') as f:
        # move the file pointer to the start of the second part
        f.seek( len(f.read())//2 )     
        # read the second part of the file
        second_part = f.read()
    # Reopen the file in write mode and delete the first half
    with open(file_path, 'w') as f:
        f.write(second_part)
    # Release the memory used by the first half
    del second_part
```

**Comment by ZuinigeRijder** [comment]
@hotting What I did, was to modify the solis.sh script, to start each day with a fresh solis.log file, by adding:

````
# new logfile every new start
echo $(date) > $logfile
````

My modified solis.sh script looks like this:
````
#!/bin/bash
# ---------------------------------------------------------------
# A script to keep the soliscloud_to_pvoutput.py script running.
# Add to your crontab to run once a day, e.g. at 05:02 and at reboot
# 2 5 * * * ~/solis/solis.sh > /dev/null
# @reboot sleep 123 && ~/solis/solis.sh > /dev/null
# ---------------------------------------------------------------
script_name=$(basename -- "$0")
cd ~/solis

now=$(date)
if pidof -x "$script_name" -o $$ >/dev/null;then
   echo "$now: $script_name already running" >> crontab.solis.log 2>&1
   exit 1
fi
echo "WARNING: $now: $script_name NOT running" >> crontab.solis.log 2>&1

# new logfile every new start
echo $(date) > solis.log

/usr/bin/python -u ~/solis/soliscloud_to_pvoutput.py >> solis.log 2>&1
````

Of course you could make the solis.sh script more intelligent, to keep more than 1 logfile or do something special to backup/halve it, before starting the python script.
